tests/checker/good/gold/dead_code_test17.scilla.gold

```diff
@@ -0,0 +1,113 @@
+{
+  "cashflow_tags": {
+    "State variables": [ { "field": "all_used", "tag": "NoInfo" } ],
+    "ADT constructors": []
+  },
+  "contract_info": {
+    "scilla_major_version": "0",
+    "vname": "Dead16",
+    "params": [
+      {
+        "vname": "all_used",
+        "type": "ByStr20 with contract field param1 : ByStr20, field param2 : ByStr20 end"
+      }
+    ],
+    "fields": [],
+    "transitions": [],
+    "procedures": [ { "vname": "use_all", "params": [] } ],
+    "events": [],
+    "ADTs": [
+      {
+        "tname": "Option",
+        "tparams": [ "'A" ],
+        "tmap": [
+          { "cname": "Some", "argtypes": [ "'A" ] },
+          { "cname": "None", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Bool",
+        "tparams": [],
+        "tmap": [
+          { "cname": "True", "argtypes": [] },
+          { "cname": "False", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Nat",
+        "tparams": [],
+        "tmap": [
+          { "cname": "Zero", "argtypes": [] },
+          { "cname": "Succ", "argtypes": [ "Nat" ] }
+        ]
+      },
+      {
+        "tname": "List",
+        "tparams": [ "'A" ],
+        "tmap": [
+          { "cname": "Cons", "argtypes": [ "'A", "List ('A)" ] },
+          { "cname": "Nil", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Pair",
+        "tparams": [ "'A", "'B" ],
+        "tmap": [ { "cname": "Pair", "argtypes": [ "'A", "'B" ] } ]
+      }
+    ]
+  },
+  "warnings": [
+    {
+      "warning_message": "Unused procedure: use_all",
+      "start_location": {
+        "file": "contracts/dead_code_test17.scilla",
+        "line": 17,
+        "column": 11
+      },
+      "end_location": { "file": "", "line": 0, "column": 0 },
+      "warning_id": 3
+    },
+    {
+      "warning_message": "Unused contract parameter: all_used",
+      "start_location": {
+        "file": "contracts/dead_code_test17.scilla",
+        "line": 10,
+        "column": 5
+      },
+      "end_location": { "file": "", "line": 0, "column": 0 },
+      "warning_id": 3
+    },
```

Comment:
I was about to complain that `all_used` is actually used, but then I realised that you are much more clever than I am. :-)

Good job.